### DISCUSSION TITLE
React + Redux might not be good

### DISCUSSION THREAD

**Discussion by Xyndra** [eythaann/Seelen-UI#236] [General]
![image](https://github.com/user-attachments/assets/b536bcd4-c7f2-4f49-9337-4688f68519c2)
[Click this for the page](https://krausest.github.io/js-framework-benchmark/2024/table_chrome_121.0.6167.86_win.html#eyJmcmFtZXdvcmtzIjpbImtleWVkL2xpdCIsImtleWVkL2xpdC1odG1sIiwia2V5ZWQvcmVhY3QtcmVkdXgiLCJrZXllZC9yZWFjdC1yZWR1eC1ob29rcyIsImtleWVkL3JlYWN0LXJlZHV4LWhvb2tzLWltbXV0YWJsZSIsImtleWVkL3JlYWN0LXp1c3RhbmQiLCJrZXllZC9zdmVsdGUiLCJrZXllZC9zdmVsdGUtY2xhc3NpYyIsImtleWVkL3ZhbmlsbGFqcyIsImtleWVkL3ZhbmlsbGFqcy0xIiwia2V5ZWQvdmFuaWxsYWpzLXdjIl0sImJlbmNobWFya3MiOlsiMDFfcnVuMWsiLCIwMl9yZXBsYWNlMWsiLCIwM191cGRhdGUxMHRoMWtfeDE2IiwiMDRfc2VsZWN0MWsiLCIwNV9zd2FwMWsiLCIwNl9yZW1vdmUtb25lLTFrIiwiMDdfY3JlYXRlMTBrIiwiMDhfY3JlYXRlMWstYWZ0ZXIxa194MiIsIjA5X2NsZWFyMWtfeDgiLCIyMV9yZWFkeS1tZW1vcnkiLCIyMl9ydW4tbWVtb3J5IiwiMjNfdXBkYXRlNS1tZW1vcnkiLCIyNV9ydW4tY2xlYXItbWVtb3J5IiwiMjZfcnVuLTEway1tZW1vcnkiLCI0MV9zaXplLXVuY29tcHJlc3NlZCIsIjQyX3NpemUtY29tcHJlc3NlZCIsIjQzX2ZpcnN0LXBhaW50Il0sImRpc3BsYXlNb2RlIjoxfQ==)

As seen in the image above, React is scoring really bad in most benchmarks. This is not really important in web development, but becomes pretty critical when you want to create native-like applications. I understand the choice for web technologies since it is pretty much the only way to get a pretty UI nowadays, however I would like to ask for reconsideration on the React framework. I am not asking for someone else to rewrite it, but I would be interested in feelings about pull request changing some of the components, for example the taskbar, into a different framework or vanilla JS. This would greatly benefit viability for this running on weaker devices that don't have a lot of RAM and CPU to spare, and would also benefit everyone else having more resources.

**Comment by eythaann** [comment]
yep I already search the same website when was making the optimizations for v2. compared with v1, v2 is really fast and lightweight, now I didn't refactor the react part cuz it is a lot of work for now, but next modules like the widgets system will be made using a lightweighter library or just vanilla js.

**Comment by eythaann** [comment]
probably I will migrating all the other components to svelte + zustand

**Reply by Xyndra** [reply]
So basically I am allowed to rewrite stuff in Svelte if I want to contribute and can't be bothered with React?

**Reply by eythaann** [reply]
hmm if you wanna colaborate with that, it needs a lot of testing, so avoid this for now, btw you know about svelte I can tag you on discord when I start the refactoring to make a entire refactor update. just focused on optimizations on UI

**Reply by Xyndra** [reply]
The problem is that I don't know how to do React since it was already known as a bad option when I started web development, so I can't really contribute anything to the existing stuff, but there are things I want to change in that React stuff so I wanted to just recode it in Svelte since I know Svelte

**Reply by eythaann** [reply]
the issue here is, or we make all, in svelte or all in react having a mixed framework will slow down the development

**Reply by eythaann** [reply]
so for now just wait until I start the refactor to svelte in a mayor version.

**Reply by Xyndra** [reply]
So basically all or nothing Svelte? If yes, it would be nice  if there was a branch to slowly start work on it
